students/DrewSmith/lessons/lesson07/assignment/linear.py

```diff
@@ -0,0 +1,215 @@
+                                                        "zip_code",
+                                                        "phone_number",
+                                                        "email"))
+            records = [record for index, record in enumerate(csv_file) if index > 0]
+            counts[1] = len(records)
+            database["customers"].insert_many(records)
+
+        with open(directory / rentals_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("product_id", "user_id"))
+            records = [record for index, record in enumerate(csv_file) if index > 0]
+            counts[2] = len(records)
+            database["rentals"].insert_many(records)
+
+    return (tuple(counts), tuple(errors))
+
+def get_prodcuts_total_record_count(connection=None, database_name=None):
+    connection = connection or MongoDBConnection()
+    database_name = database_name or config.DATABASE_NAME
+
+    with connection:
+        database = connection.connection.get_database(name=database_name)
+        return sum((database[collection].count() for collection in ["products", "customers", "rentals"]))
+
+
+# Customers (sqlite)
+def convert_csv(file_path):
+    '''
+    Generator for CSV file formatted for Customer model data
+
+    :param file_path: file path for the CSV file
+    :yield: CSV line dictionary formatted for Customer model
+    '''
+    fields = ('customer_id', 'first_name', 'last_name', 'home_address',
+              'phone_number', 'email_address', 'status', 'credit_limit')
+    with open(file_path, 'r') as file:
+        csv_file = csv.DictReader(file, fieldnames=fields)
+        for index, row in enumerate(csv_file):
+            if index == 0:
+                continue
+
+            row['phone_number'] = "".join(
+                char for char in row['phone_number'] if char not in  '.()- ')
+            row['status'] = bool(row['status'].lower() == 'active')
+            row['credit_limit'] = float(row['credit_limit'])
+            yield row
+
+def bulk_add_customers(customers):
+    '''
+    Adds a bulk amount of customers
+
+    :param customers: sequence of dictionaries, expects keys:
+        customer_id, first_name, last_name, home_address,phone_number, email_address,
+        status, credit_limit
+    '''
+    records_added = 0
+    batch_size = 120
+    customer_list = list()
+    database = cc.get_database()
+    for index, customer in enumerate(customers, start=1):
+        try:
+            customer['credit_limit'] = float(customer['credit_limit'])
+        except ValueError as err:
+            logging.error(err)
+            raise ValueError(f"Invalid value: record {index}: " \
+                f"invalid credit_limit: '{customer['credit_limit']}'")
+
+        if not isinstance(customer['status'], bool):
+            text = f"Invalid value: record {index}: status is not a bool: '{customer['status']}'"
+            logging.error(text)
+            raise ValueError(text)
```

Comment:
I would consider putting the error checking in a separate function to make this function smaller and more readable.